src/components/Modals/ManageCollectionRoleModal/ManageCollectionRoleModal.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react'
+import { ModalNavigation, ModalActions, Button } from 'decentraland-ui'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import Modal from 'decentraland-dapps/dist/containers/Modal'
+
+import { isValid } from 'lib/address'
+import { getSaleAddress, setOnSale, isOnSale } from 'modules/collection/utils'
+import { Access, RoleType } from 'modules/collection/types'
+import { isEqual } from 'lib/address'
+import Role from './Role'
+import EmptyRole from './EmptyRole'
+import { Props, State } from './ManageCollectionRoleModal.types'
+import './ManageCollectionRoleModal.css'
+
+export default class ManageCollectionRoleModal extends React.PureComponent<Props, State> {
+  state: State = { roles: this.getStartRoles() }
+
+  getStartRoles() {
+    const { metadata } = this.props
+    const roles = new Set(metadata.roles)
+    roles.delete(this.getSaleAddress())
+    return Array.from(roles)
+  }
+
+  handleAddNewRole = () => {
+    const { roles } = this.state
+    this.setState({
+      roles: [...roles, undefined]
+    })
+  }
+
+  handleAddRole = (index: number, role: string) => {
+    if (this.isValidRole(role)) {
+      const roles = this.removeRoleAtIndex(index)
+      roles.push(role.toLowerCase())
+
+      this.setState({ roles })
+    }
+  }
+
+  handleRemoveRole = (role: string) => {
+    const { roles } = this.state
+    this.setState({
+      roles: roles.filter(_role => _role !== role)
+    })
+  }
+
+  handleCancelNew = (index: number) => {
+    this.setState({
+      roles: this.removeRoleAtIndex(index)
+    })
+  }
+
+  handleSubmit = () => {
+    const { collection, metadata } = this.props
+    const { roles } = this.state
+    const originalRoles = metadata.roles
```

Comment:
If the collection is on sale we are "removing" the store address because it is not at `this.state.roles` but at `originalRoles`, right?

Reply:
I guess that the same is happening when trying to add collaborators